geps/gep-2907/index.md

```diff
@@ -0,0 +1,109 @@
+# GEP-2907: TLS Configuration Placement and Terminology
+
+* Issue: [#2907](https://github.com/kubernetes-sigs/gateway-api/issues/2907)
+* Status: Memorandum
+
+## TLDR
+This GEP aims to define high level TLS terminology and structure within Gateway
+API to ensure that all our independent proposals related to TLS result in a
+coherent set of APIs.
+
+## Goals
+* Define high-level terminology for how we refer to TLS in Gateway API.
+* Define top level fields where TLS configuration can live.
+
+## Non-Goals
+* Add or change fields directly. (This may inspire changes in other GEPs
+  though).
+* Commit to including specific parts of TLS configuration in Gateway API. (This
+  is merely to provide space for future configuration, not a commitment that we
+  will add it to the API.)
+
+## Introduction
+
+There are essentially 4 different "buckets" that could be relevant to TLS
+configuration in Gateway API:
+
+```mermaid
+flowchart LR
+    Frontend -.->|1. Client Cert| Gateway
+    Gateway -.->|2. Server Cert| Frontend
+    Gateway -.->|3. Client Cert| Backend
+    Backend -.->|4. Server Cert| Gateway
+```
+
+### Where it can fit
+We have three different places where this configuration could live:
+
+#### A. Gateways
+We can add config that applies to an entire Gateway. Config attached at this
+layer will apply to everything behind a single VIP.
+
+#### B. Gateway Listeners
+Each Gateway contains multiple “Listeners”. Each Listener in a Gateway must have
+a unique combination of protocol, port, and hostname (SNI in the case of HTTPS).
+TLS configuration attached at this layer needs to be isolated via SNI.
+
+#### C. BackendTLSPolicy
+This policy allows us to attach unique TLS configuration per Backend. Depending
+on the organization, this policy may be owned by the application owner or the LB
+admin. Note that this configuration will be used by all Gateways (potentially
+from different implementations) that are connecting to the backend.
+
+## Proposed Buckets
+Note that this does not represent any form of commitment that any of these
+fields or concepts will exist within Gateway API. If they did, they would follow
+the following naming structure though:
+
+### 1. Validate Client Certificate provided by Frontend
+
+| Proposed Placement | Name | Status |
+|-|-|-|
+| Gateway Listener | `Listener.TLS.FrontendValidation` | Proposed |
```

Comment:
did you mean `Listener.TLS.Frontend.Validation` ?

Reply:
Nope, this was intentional. Part of the idea behind this GEP is that `Listener.TLS` is really entirely focused on 1+2 - the bits of TLS between frontend and Gateway. That means that if we were to add any additional TLS configuration in the `Listerner.TLS` struct, it would tied to that limited scope, and thus we don't really need a separate `Frontend` struct.

One could make an argument that `Listener.TLS.Validation` should be the field name here to avoid any ambiguity, but in this specific context, I think that it's probably helpful to specifically spell out `frontend`.

Reply:
I think that if you include basically the above comment into the text of the GEP, we should be good here.

Reply:
Copied nearly verbatim, thanks!